promise.d.ts

```diff
@@ -42,6 +44,8 @@ export interface PreparedStatementInfo {
 export interface Connection extends QueryableAndExecutableBase {
   config: ConnectionOptions;
 
+  connection: CoreConnection;
```

Comment:
This connection field is defined on the pool Connection, so this is now available everywhere pool connection is used.